lib/Analysis/VectorUtils2.cpp

```diff
@@ -0,0 +1,72 @@
+//===----------- VectorUtils2.cpp - findScalarElement function -----------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines vectorizer utility function findScalarElement.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Analysis/VectorUtils.h"
+#include "llvm/IR/PatternMatch.h"
+#include "llvm/IR/Value.h"
+
+// NOTE: Splitting this function from VectorUtils.cpp into a separate file 
+// makes dxilconv.dll 121kB smaller (x86 release, compiler optimization for size).
```

Comment:
I would have expected to see file-specific commentary above the includes with the "This file defines" line.

Reply:
I've moved it up.